### ISSUE TITLE
Runtime error segmentation fault core dumped

### ISSUE THREAD

**Issue by thebadgateway** [lhy0403/Qv2ray#34]
The error was on the previous version.
Occurs when changing the language.
[IMG](https://pixs.ru/images/2019/09/03/3-SEN-2019--054301.png)

**Comment by lhy0403** [comment]
A cause of the issue has been addressed, will fix tomorrow,

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by lhy0403** [comment]
Can you try to recreate the same issue using the latest release?

**Comment by thebadgateway** [comment]
New version:
```
sony@sony-desktop:~/test$ git clone --recursive https://github.com/lhy0403/Qv2ray && cd Qv2ray
Клонирование в «Qv2ray»…
remote: Enumerating objects: 19, done.
remote: Counting objects: 100% (19/19), done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 1452 (delta 3), reused 10 (delta 2), pack-reused 1433
Получение объектов: 100% (1452/1452), 4.43 MiB | 2.62 MiB/s, готово.
Определение изменений: 100% (1024/1024), готово.
Подмодуль «3rdparty/x2struct» (https://github.com/xyz347/x2struct) зарегистрирован по пути «3rdparty/x2struct»
Клонирование в «/home/sony/test/Qv2ray/3rdparty/x2struct»…
remote: Enumerating objects: 45, done.        
remote: Counting objects: 100% (45/45), done.        
remote: Compressing objects: 100% (32/32), done.        
remote: Total 933 (delta 29), reused 27 (delta 13), pack-reused 888        
Получение объектов: 100% (933/933), 2.72 MiB | 1.95 MiB/s, готово.
Определение изменений: 100% (501/501), готово.
Подмодуль по пути «3rdparty/x2struct»: забрано состояние «084c3cd183ae2450473befa53e987a0ed347a6ff»
sony@sony-desktop:~/test/Qv2ray$ mkdir build && cd build
sony@sony-desktop:~/test/Qv2ray/build$ qmake ../
Info: creating stash file /home/sony/test/Qv2ray/build/.qmake.stash
RCC: Error in '../resources.qrc': Cannot find file 'translations/zh-CN.qm'
RCC: Error in '../resources.qrc': Cannot find file 'translations/en-US.qm'
```
After which I rename the files in folder translations to *.qm
Make is successful and when I start, I see an [error](https://pixs.ru/images/2019/09/03/3-SEN-2019--055642.png)

**Comment by thebadgateway** [comment]
> Can you try to recreate the same issue using the latest release?

The new version is exactly the same.

**Comment by lhy0403** [comment]
So let's say you are trying to build from the source right?

**Comment by lhy0403** [comment]
If you build it from the source, you should not rename the `.ts` files in the `translations` folder to `.qm` but run `lrelease ./Qv2ray.pro` to let Qt generate the `.qm`-named the binary version of "translation files".

Then try to build again.

The docs are [here](https://github.com/lhy0403/Qv2ray/blob/master/README.md#linux--macos). Unfortunately, it's written in Chinese, but the English version will come soon.

**Comment by thebadgateway** [comment]
> If you build it from the source, you should not rename the `.ts` files in the `translations` folder to `.qm` but run `lrelease ./Qv2ray.pro` to let Qt generate the `.qm`-named the binary version of "translation files".
> 
> Then try to build again.
> 
> The docs are [here](https://github.com/lhy0403/Qv2ray/blob/master/README.md#linux--macos). Unfortunately, it's written in Chinese, but the English version will come soon.

Yes, I know that I forgot team `lrelease ./Qv2ray.pro` last time.
The new version also gives an error. [PIC](https://pixs.ru/images/2019/09/04/4-SEN-2019--064834.png)

**Comment by thebadgateway** [comment]
Please add the field "Name" [PIC](https://pixs.ru/images/2019/09/04/Desktop.png)

**Comment by lhy0403** [comment]
> Please add the field "Name" [PIC](https://pixs.ru/images/2019/09/04/Desktop.png)

Sure, This will be available in a later version.

**Comment by lhy0403** [comment]
> Yes, I know that I forgot command `lrelease ./Qv2ray.pro` last time.
> The new version also gives an error. [PIC](https://pixs.ru/images/2019/09/04/4-SEN-2019--064834.png)

I cannot figure out what's going wrong here... We only support Chinese and English currently.
It's more likely caused by a missing of Chinese Fonts(maybe?), 

However...
I'd prefer to say that's a Qt Bug....
I'll try my best to understand the cause of your issue at the moment

**Comment by thebadgateway** [comment]
> > Yes, I know that I forgot command `lrelease ./Qv2ray.pro` last time.
> > The new version also gives an error. [PIC](https://pixs.ru/images/2019/09/04/4-SEN-2019--064834.png)
> 
> I cannot figure out what's going wrong here... We only support Chinese and English currently.
> It's more likely caused by a missing of Chinese Fonts(maybe?),
> 
> However...
> I'd prefer to say that's a Qt Bug....
> I'll try my best to find the cause of your issue at the moment

I can give access to the teamviewer if you need to.

**Comment by lhy0403** [comment]
Yes that'll be great

**Comment by lhy0403** [comment]
[patch1.txt](https://github.com/lhy0403/Qv2ray/files/3574432/patch1.txt)

Well, before that, can you try this patch please? 
This patch changed the "just-in-time retranslation" function to another trigger.

**Comment by thebadgateway** [comment]
> [patch1.txt](https://github.com/lhy0403/Qv2ray/files/3574432/patch1.txt)
> 
> Well, before that, can you try this patch please?
> This patch changed the "just-in-time retranslation" function to another trigger.

I need instruction to patch1.txt

**Comment by lhy0403** [comment]
Well it's not very difficult, download the patch file and then simply run 
`git apply patch1.txt`
 in the Qv2ray project directory

**Comment by thebadgateway** [comment]
```
sony@sony-desktop:~$ cd test/1/Qv2ray/
sony@sony-desktop:~/test/1/Qv2ray$ git apply patch1.txt
sony@sony-desktop:~/test/1/Qv2ray$ cd build/
sony@sony-desktop:~/test/1/Qv2ray/build$ ./Qv2ray 
```
To no avail.
I tried the same make application. Doesn't work either.

**Comment by lhy0403** [comment]
Emmm after applying the patch, you need to rebuild the executable using `make` again.

**Comment by thebadgateway** [comment]
> Emmm after applying the patch, you need to rebuild the executable using `make` again.

That's exactly what I did

**Comment by lhy0403** [comment]
> > Emmm after applying the patch, you need to rebuild the executable using `make` again.
> 
> That's exactly what I did

It seems to be unbelievably strange.... 
Which distro are you currently using, or can you try installing a Chinese Font? (Although I don't think that helps....)
- You confirm that the latest AppImage did the same right?

**Comment by lhy0403** [comment]
Another trail is to change the config file directly located at `$HOME/.qv2ray/Qv2ray.conf`
You may want to:

- Close Qv2ray
- Change the `"language": "zh-CN",` to `"language": "en-US",` 
- Start Qv2ray again and see if the translation loads correctly.

**Comment by thebadgateway** [comment]
I think I have a Chinese font. I can’t switch from Chinese to English. BUT! If I rename the files en-US.ts <-> zh-CN.ts before installation and make the application, then it works in English.

**Comment by thebadgateway** [comment]
> Another trail is to change the config file directly located at `$HOME/.qv2ray/Qv2ray.conf`
> You may want to:
> 
> * Close Qv2ray
> * Change the `"language": "zh-CN",` to `"language": "en-US",`
> * Start Qv2ray again and see if the English Translation loads correctly.
> 
> Updated: This helps me see if the issue is caused by the translation itself or the UI behavior

The application started in English and works. But if you open the settings, then it freezes.

**Comment by lhy0403** [comment]
> I think I have a Chinese font. I can’t switch from Chinese to English. BUT! If I rename the files en-US.ts <-> zh-CN.ts before installation and make the application, then it works in English.

Emmm at least that tells us the cause is not from a language translation file......
One possible solution is disabling the "just-in-time retranslation" feature currently....

**Comment by lhy0403** [comment]
I'll push a commit and see if that works for you soon.

NOTE: You may need to restart the application after changing the language from now on. (As a temporary solution for this strange bug)

**Comment by lhy0403** [comment]
You may want to checkout the latest commit on the `dev` branch
using: 
`git reset master --hard` <- to discard whatever you did in the current working directory
`git pull` <- pull out the latest update
`git checkout dev` <- and change to the dev branch.
run qmake and make to see if it's solved.

**Comment by thebadgateway** [comment]
> You may want to checkout the latest commit on the `dev` branch
> using:
> `git reset master --hard` <- to discard whatever you did in the current working directory
> `git pull` <- pull out the latest update
> `git checkout dev` <- and change to the dev branch.
> run qmake and make to see if it's solved.

It works! There is not any errors.

**Comment by lhy0403** [comment]
OK, then this issue will be marked as solved and closed.

**Comment by lhy0403** [comment]
For future refrence:
The problem was introduced by #20 , which firstly asked for the "just-in-time re-translation". 
Due to the bugfix, this feature has been disabled.

**Comment by lhy0403** [comment]
Hello sir Seems that you speak Russian
Could you contribute to translate the project in Crowdin Service here?
https://crowdin.com/project/qv2ray
It will be very helpful.

**Comment by thebadgateway** [comment]
> Hello sir Seems that you speak Russian
> Could you contribute to translate the project in Crowdin Service here?
> https://crowdin.com/project/qv2ray
> It will be very helpful.

Im done

**Comment by lhy0403** [comment]
Thanks, and can you please provide your name that I can mention the contributors?
is `thebadgateway` ok for that ?

**Comment by thebadgateway** [comment]
> Thanks, and can you please provide your name that I can mention the contributors?
> is `thebadgateway` ok for that ?

Yes of course. The only thing you need will be to test the translation program. Some words can be long and you can pick up similar ones in meaning, but with a shorter length. And without a program, some words have two meanings. Only after testing can I give an accurate translation.

**Comment by lhy0403** [comment]
Thanks really a lot for your contributions!
However, I didn't understand what you mean, do you want to have a translated Qv2ray application for testing?

**Comment by thebadgateway** [comment]
I meant that I would need to see the translated application. And then I can correct some words because they can have two meanings. In general, nothing serious, I will write a issure if I see a mistake in my translation.

**Comment by lhy0403** [comment]
There are two possible methods to get your translations.

## Option 1
One is you can download the *.ts file in the Crowdin Platform
You can download the translations here:
![image](https://user-images.githubusercontent.com/18734999/65014321-e559f600-d950-11e9-8ac0-6032db74903c.png)
- For this, you may need to clone the Repo and manually replace the .ts file in the `translations` folder. 
- And then, build the Qv2ray.
- It's fast, you can do whenever you want, but it's really quite complicated.

## Option 2

- There's another option, you can wait for the Crowdin Service writing a Pull Request [HERE](https://github.com/lhy0403/Qv2ray/pulls）
- The Automatic Build System will trigger a build containing your translations into an AppImage. Which you can download it and test. 
![image](https://user-images.githubusercontent.com/18734999/65014754-1edf3100-d952-11e9-8aa5-5a123d091854.png)
Clocking 'details' leads you to the status page: 
![image](https://user-images.githubusercontent.com/18734999/65014826-5948ce00-d952-11e9-8970-f07742beb0db.png)
The binary can be obtained here
The second one is the build system (currently not generating AppImage because it's not properly configured *yet*)

-----------

Actrually I, personally prefer to build myself, as **even I don't know how long it takes** for the Crowdin to sync translations into the Github PR

<I'll write a proper doc for this soon>

**Comment by lhy0403** [comment]
Ah, I nearly forgot that, You can try to use Qt Linguist to open that TS file and do some translations!
![image](https://user-images.githubusercontent.com/18734999/65015080-1804ee00-d953-11e9-9bc4-c6d4f05d6197.png)
It's soooooo useful to test some UI translations, once you have done it, click save and upload to the Crowdin,

**Comment by thebadgateway** [comment]
The new version does not work

`:~/test/2/Qv2ray/build$ ./Qv2ray 
[LICENCE]: 
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions.

Qv2ray Current Developer Copyright (C) 2019 Leroy.H.Y (@lhy0403)
Hv2ray Initial Idea and Designs Copyright (C) 2019 Hork (@aliyuchang33)
Hv2ray/Qv2ray HTTP Request Helper (partial) Copyright 2019 (C) SOneWinstone (@SoneWinstone)
Qv2ray ArtWork Done By ArielAxionL (@axionl)

Qv2ray v2.0 running on Ubuntu 18.04.3 LTS x86_64

[UI]: FAILED to find any translations, THIS IS A BUILD ERROR.
QWidget: Must construct a QApplication before a QWidget`

**Comment by lhy0403** [comment]
can you please open another issue and we can discuss there ?
